include/picongpu/particles/atomicPhysics2/kernel/ChooseTransition_BoundBound.kernel

```diff
@@ -225,22 +231,15 @@ namespace picongpu::particles::atomicPhysics2::kernel
                             // test for each bin separately
                             for(int binIndex = 0; binIndex < T_Histogram::numberBins; ++binIndex)
                             {
-                                // eV
-                                float_X const energy = electronHistogram.getBinEnergy(u32(binIndex));
-                                // eV
-                                float_X const binWidth = electronHistogram.getBinWidth(u32(binIndex));
-                                // 1/(UNIT_LENGTH^3 * eV)
-                                float_X const density
-                                    = electronHistogram.getBinWeight0(u32(binIndex)) / volumeScalingFactor / binWidth;
-
+                                float_X const density = cachedHistogram.density[binIndex];
                                 // 1/UNIT_TIME
                                 float_X const rateTransition = picongpu::particles::atomicPhysics2::rateCalculation::
                                     BoundBoundTransitionRates<T_n_max>::template rateCollisionalBoundBoundTransition<
                                         T_AtomicStateDataDataBox,
                                         T_BoundBoundTransitionDataBox,
                                         isUpward>(
-                                        energy,
-                                        binWidth,
+                                        cachedHistogram.energy[binIndex],
+                                        cachedHistogram.binWidth[binIndex],
                                         density,
```

Comment:
why not an in place call?

Reply:
density is used in the debug message too